### PR TITLE
CASM-3727: Added an output parameter to the iuf-base-template.

### PR DESCRIPTION
**Pull request by jsollom-hpe**
## Summary and Scope

* Added an output parameter to the iuf-base-template.
* Added an input parameter to define the file wherein the script’s output will be stored within the container.
* The name of the file where the script should store its output is communicated via an environment variable: SCRIPT_OUTPUT_FILE.
* Added descriptions to the input parameters.

## Issues and Related PRs

_List and characterize relationship to Jira/Github issues and other pull requests. Be sure to list dependencies._

* Resolves [CASM-3727]

## Testing

_List the environments in which these changes were tested._

### Tested on:

  * `Frigg`

### Test description:

I ran a workflow where I uploaded a Loftsman manifest and made sure that manifest was recorded in the product catalog under the correct product.

Were the install/upgrade-based validation checks/tests run (goss tests/install-validation doc)?
Were continuous integration tests run? If not, why? No
Was upgrade tested? If not, why? No
Was downgrade tested? If not, why? No
Were new tests (or test issues/Jiras) created for this change? No, this is new code.

## Risks and Mitigations
Low

## Pull Request Checklist

- [x] Version number(s) incremented, if applicable
- [x] Copyrights updated
- [x] License file intact
- [x] Target branch correct
- [x] CHANGELOG.md updated
- [x] Testing is appropriate and complete, if applicable
- [x] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

**Comment by jsollom-hpe** [comment]
I don't know if I need to update the Helm chart version in cray-nls.

**Comment by ekamin-hpe** [comment]
> I don't know if I need to update the Helm chart version in cray-nls.

Unfortunately, I believe you do. The container image version (same value found in `.version`) gets updated in `cray-nls/charts/v1.0/cray-nls/values.yaml`, and the overall chart version gets updated in `cray-nls/charts/v1.0/cray-nls/Chart.yaml`.

Another value I think needs updating is the "version" field on line 29 of `iufBase.template.argo.yaml` per recent slack thread: https://cray.slack.com/archives/C04593DNN3U/p1675113417365639

File: src/api/argo-templates/base/iufBase.template.argo.yaml [hunk 1]
```diff
@@ -34,7 +34,26 @@ spec:
...
        parameters:
          - name: scriptContent
+           description: >
+             The content of the script that will be run
          - name: dryRun
+           description: >
+             A flag indicating whether to run the script or not.
+             True means do not run the script merely output it to a file.
+         - name: script_output_file
+           value: "/tmp/script_output"
```
Comment (heemstra) [inline src/api/argo-templates/base/iufBase.template.argo.yaml:44]: I don't have a good feel for whether this is an issue, so I'll just ask: any concerns about this being overwritten from one use of the template to the next?

**Comment by heemstra** [comment]
> I don't know if I need to update the Helm chart version in cray-nls.

You might want to chat with @bo-quan.  He created a separate repo for the chart, IIRC.

File: src/api/argo-templates/base/iufBase.template.argo.yaml (hunk 1 above)
Comment (ekamin-hpe) [reply src/api/argo-templates/base/iufBase.template.argo.yaml:44]: That file should only live in ephemeral container storage so I think we're good.

**Review by ekamin-hpe — APPROVED** [review]
The mod looks good to me. I have some questions about versioning - I thought it was done just by updating the cray-nls values like I did in #144, getting confused about how the iuf chart fits in.

**Comment by jsollom-hpe** [comment]
> > I don't know if I need to update the Helm chart version in cray-nls.
> 
> You might want to chat with @bo-quan. He created a separate repo for the chart, IIRC.

Talk with @bo-quan about whether there is actually a separate repository containing this Helm chart or not. The chart still appears in this repository.

**Review by heemstra — APPROVED** [review]

**Review by ekamin-hpe — APPROVED** [review]

**Review by heemstra — APPROVED** [review]

**Review by bo-quan — CHANGES_REQUESTED** [review]
Please use another repo cray-nls-charts for chart changes (i.e., Chart.yaml, and values.yaml)

Done. I did so in this PR:
https://github.com/Cray-HPE/cray-nls-charts/pull/23

**Review by bo-quan — APPROVED** [review]

**Pull request merged**